eth-node/core/types/transaction.go

```diff
@@ -0,0 +1,39 @@
+package types
+
+import (
+	"github.com/status-im/status-go/eth-node/types"
+	"math/big"
+)
+
+type Message struct {
+	to         *types.Address
+	from       types.Address
+	nonce      uint64
+	amount     *big.Int
+	gasLimit   uint64
+	gasPrice   *big.Int
+	data       []byte
+	checkNonce bool
+}
+
+func NewMessage(from types.Address, to *types.Address, nonce uint64, amount *big.Int, gasLimit uint64, gasPrice *big.Int, data []byte, checkNonce bool) Message {
```

Comment:
Following the convention, `New` should return a pointer. You won't need this at all if you export `Message` fields.

Reply:
these are copied from `geth` should we not keep them the same for consistency?